### PR TITLE
fix(dockerfile): Run as non-root.

### PR DESCRIPTION
**Pull request by mikaeld**
Because

- Running the application as root inside the container can pose security risks, and it's best practice to run containers as a non-root user. GCPv2 security requirements proscribes running containers as root.

This commit

- Modifies the Dockerfile to create and use a non-root user with UID 10001.
- Updates ownership of necessary directories and files to ensure the non-root user can access and modify them.
- Switches the running user to the non-root user `appuser` after environment setup.

# Related PR
https://github.com/mozilla-it/webservices-infra/pull/2741

### CODE REVIEW

File: experimenter/Dockerfile
Comment (mikaeld) [inline experimenter/Dockerfile:129]: Is poetry actually required in the final stage?

Comment (jaredlockhart) [reply experimenter/Dockerfile:129]: Hrmm good question.  It shouldn't be, let me confirm.

**Comment by jaredlockhart** [comment]
@mikaeld Looks like we have the same hard coded celery paths in the prod infra as well: 

https://github.com/mozilla-services/cloudops-infra/blob/ed5ac24f6c744c3d5f9b941bf58217f1f606a833/projects/experimenter/k8s/charts/experimenter/templates/deployment-beat.yaml#L47-L52

We'll have to remove those before we merge this so celery can start after we merge this.  Do you want to update that there?

**Comment by mikaeld** [comment]
@jaredlockhart  Thanks for helping on this! I will file a PR to change the hard coded path in `cloudops-infra` and we should be good to deploy tomorrow!

**Review by yashikakhurana — APPROVED** [review]
Looks good

**Pull request merged**
